pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -645,7 +646,7 @@ func (c *controller) reconcileMachineHealth(ctx context.Context, machine *v1alph
 			}
 			cloneDirty = true
 
-		} else if c.isHealthy(clone) && clone.Status.CurrentStatus.Phase != v1alpha1.MachineRunning {
+		} else if c.isHealthy(clone, node) && clone.Status.CurrentStatus.Phase != v1alpha1.MachineRunning {
```

Comment:
With the suggested way we won't block the `Unknown` machines(primarily) from turning to `Ready` because of the taint.
```suggestion
		} else if c.isHealthy(clone, node) {
		      if clone.Status.CurrentStatus.Phase == v1alpha1.MachinePending{
		       if !c.criticalComponentsNotReadyTaintPresent(node) {
		      ... // turn to Running
		      } else if clone.Status.CurrentStatus.Phase != v1alpha1.MachineRunning{
		      ...// turn to Running
		      }
		}
```